components/support/utils/src/main/java/mozilla/components/support/utils/URLStringUtils.kt

```diff
@@ -90,8 +90,9 @@ object URLStringUtils {
         // -x.com
         // http://www-.com
         // www.c-c-
+        // https://disposable-wide4--solidsoft-blog-disposable.netlify.app/ (fenix/issues/10747)
         // 3-3
-        Pattern.compile("^\\s*(\\w+-)*\\w+(://[/]*|:|\\.)(\\w+-)*\\w+([\\S&&[^\\w-]]\\S*)?\\s*$", flags)
+        Pattern.compile("^\\s*(\\w+-)*\\w+(://[/]*|:|\\.)(\\w+-+)*\\w+([\\S&&[^\\w-]]\\S*)?\\s*$", flags)
```

Comment:
We should also update the first `(\\w+-)` group here to `(\\w+-+)`, same as the second one, because now this is a valid URL:

`http://abc-abc--abc-abc.com`

but this isn't

`abc-abc--abc-abc.com`

Seem my comment above :)

So, the regex becomes:
```suggestion
        Pattern.compile("^\\s*(\\w+-+)*\\w+(://[/]*|:|\\.)(\\w+-+)*\\w+([\\S&&[^\\w-]]\\S*)?\\s*$", flags)
```